miio/fan_miot.py

```diff
@@ -525,3 +547,312 @@ def delay_off(self, minutes: int):
+
+    @property
+    def powersupply_attached(self) -> bool:
+        """True is power supply is attached."""
+        return self.data["powersupply_attached"]
+
+    @property
+    def speed_rpm(self) -> int:
+        """Fan rotations per minute."""
+        return self.data["speed_rpm"]
+
+    @property
+    def oscillate(self) -> bool:
+        """True if oscillation is enabled."""
+        return self.data["swing_mode"]
+
+    @property
+    def angle(self) -> int:
+        """Oscillation angle."""
+        return self.data["swing_mode_angle"]
+
+    @property
+    def temperature(self) -> Any:
+        """Air temperature (degree celsius)."""
+        return self.data["temperature"]
+
+
+class FanZA5(MiotDevice):
+    mapping = MIOT_MAPPING[MODEL_FAN_ZA5]
+
+    def __init__(
+        self,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+        model: str = MODEL_FAN_ZA5,
+    ) -> None:
+        super().__init__(ip, token, start_id, debug, lazy_discover)
+        self.model = model
+
+    @command(
+        default_output=format_output(
+            "",
+            "Angle: {result.angle}\n"
+            "Battery Supported: {result.battery_supported}\n"
+            "Buttons Pressed: {result.buttons_pressed}\n"
+            "Buzzer: {result.buzzer}\n"
+            "Child Lock: {result.child_lock}\n"
+            "Delay Off Countdown: {result.delay_off_countdown}\n"
+            "Fan Level: {result.fan_level}\n"
+            "Fan Speed: {result.fan_speed}\n"
+            "Humidity: {result.humidity}\n"
+            "Ionizer: {result.ionizer}\n"
+            "Light: {result.light}\n"
+            "Mode: {result.mode.name}\n"
+            "Oscillate: {result.oscillate}\n"
+            "Power: {result.power}\n"
+            "Powersupply Attached: {result.powersupply_attached}\n"
+            "Speed RPM: {result.speed_rpm}\n"
+            "Temperature: {result.temperature}\n",
+        )
+    )
+    def status(self):
+        """Retrieve properties."""
+        return FanStatusZA5(
+            {
+                prop["did"]: prop["value"] if prop["code"] == 0 else None
+                for prop in self.get_properties_for_mapping()
+            }
+        )
+
+    @command(default_output=format_output("Powering on"))
+    def on(self):
+        """Power on."""
+        return self.set_property("power", True)
+
+    @command(default_output=format_output("Powering off"))
+    def off(self):
+        """Power off."""
+        return self.set_property("power", False)
+
+    @command(default_output=format_output("Turning ionizer on"))
+    def ionizer_on(self):
+        """Turn ionizer on."""
+        return self.set_property("anion", True)
+
+    @command(default_output=format_output("Turning ionizer off"))
+    def ionizer_off(self):
+        """Turn ionizer off."""
+        return self.set_property("anion", False)
```

Comment:
Use `ionizer(self, on: bool)` just like for oscillate & buzzer to avoid separate command for toggling.

Reply:
Done.